project/generate-dockerfile.sh

```diff
@@ -0,0 +1,255 @@
+EOF
+if [ -z $GCCGO ]; then
+	cat <<-'EOF'
+		RUN cd /usr/local/lvm2 \
+			&& ./configure --enable-static_link \
+			&& make device-mapper \
+			&& make install_device-mapper
+			# see https://git.fedorahosted.org/cgit/lvm2.git/tree/INSTALL
+	EOF
+else
+	cat <<-'EOF'
+		RUN rm -rf /usr/local/lvm2
+		RUN git clone --no-checkout git://git.fedorahosted.org/git/lvm2.git /usr/local/lvm2 && cd /usr/local/lvm2 && git checkout -q v2_02_103
+		RUN curl -o /usr/local/lvm2/autoconf/config.guess 'http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD'
+		RUN curl -o /usr/local/lvm2/autoconf/config.sub 'http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD'
+		# Compile and install lvm2
+		RUN cd /usr/local/lvm2 \
+			&& ./configure --enable-static_link \
+			&& make device-mapper \
+			&& make install_device-mapper
+	EOF
+fi
+cat <<-'EOF'
+
+# Install lxc
+ENV LXC_VERSION 1.0.7
+RUN mkdir -p /usr/src/lxc \
+        && curl -sSL https://linuxcontainers.org/downloads/lxc/lxc-${LXC_VERSION}.tar.gz | tar -v -C /usr/src/lxc/ -xz --strip-components=1
+RUN cd /usr/src/lxc \
+        && ./configure \
+        && make \
+        && make install \
+        && ldconfig
+
+EOF
+if [ "$GCCGO" == "true" ]; then 
+	cat <<-'EOF'
+		ENV     GOPATH  /go:/go/src/github.com/docker/docker/vendor
+	EOF
+else
+	cat <<-'EOF'
+		# Install Go
+		ENV GO_VERSION 1.4.1
+		RUN curl -sSL https://golang.org/dl/go${GO_VERSION}.src.tar.gz | tar -v -C /usr/local -xz \
+			&& mkdir -p /go/bin
+		ENV PATH /go/bin:/usr/local/go/bin:$PATH
+		ENV GOPATH /go:/go/src/github.com/docker/docker/vendor
+		RUN cd /usr/local/go/src && ./make.bash --no-clean 2>&1
+    
+		# Compile Go for cross compilation
+		ENV DOCKER_CROSSPLATFORMS \
+			linux/386 linux/arm \
+			darwin/amd64 darwin/386 \
+			freebsd/amd64 freebsd/386 freebsd/arm
+
+		# TODO when https://jenkins.dockerproject.com/job/Windows/ is green, add windows back to the list above
+		#       windows/amd64 windows/386
+
+		# (set an explicit GOARM of 5 for maximum compatibility)
+		ENV GOARM 5
+		RUN cd /usr/local/go/src \
+			&& set -x \
+			&& for platform in $DOCKER_CROSSPLATFORMS; do \
+			GOOS=${platform%/*} \
+			GOARCH=${platform##*/} \
+			./make.bash --no-clean 2>&1; \
+		done
+
+		# We still support compiling with older Go, so need to grab older "gofmt"
+		ENV GOFMT_VERSION 1.3.3
+		RUN curl -sSL https://storage.googleapis.com/golang/go${GOFMT_VERSION}.$(go env GOOS)-$(go env GOARCH).tar.gz | tar -C /go/bin -xz --strip-components=2 go/bin/gofmt
+	EOF
+fi
+cat <<-'EOF'
+
+# Grab Go's cover tool for dead-simple code coverage testing
+RUN go get golang.org/x/tools/cmd/cover
+
+# TODO replace FPM with some very minimal debhelper stuff
+EOF
+if [ "$GCCGO" == "true" ]; then
+	cat <<-'EOF'
+		RUN apt-get install -y libffi-dev
+	EOF
+fi
+cat <<-'EOF'
+RUN gem install --no-rdoc --no-ri fpm 
+
+# Get the "busybox" image source so we can build locally instead of pulling
+EOF
```

Comment:
I'll leave it to @tianon to judge, but I would add newlines before each `if` and after each `fi` for readibility.